### PR TITLE
DevDocs: Add UserItem component

### PR DESCRIPTION
**Pull request by andfinally**
This PR partially addresses #24136 - a lot of components and blocks aren't listed in DevDocs.

* Added `UserItem` component to DevDocs and DevDocs playground.
* Reordered import statements in `playground.jsx`.

cc @scruffian

### CODE REVIEW

File: client/components/user/docs/example.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import UserItem from '../index';
+import user from 'lib/user';
+
+function UserItemExample() {
+	const currentUser = user().get();
```
Comment (rodrigoi) [inline client/components/user/docs/example.jsx]: maybe it'll be better to user the `getCurrentUser` selector and connect `UserItemExample` to the store to reflect the current data access practices of Calypso.

Comment (rodrigoi) [reply client/components/user/docs/example.jsx]: This may be a good example: https://github.com/Automattic/wp-calypso/blob/master/client/blocks/author-selector/docs/example.jsx

File: client/devdocs/design/playground.jsx [hunk 2]
```diff
@@ -88,6 +88,9 @@ import JetpackColophon from 'components/jetpack-colophon';
 import JetpackLogo from 'components/jetpack-logo';
 import LanguagePicker from 'components/language-picker';
 import ListEnd from 'components/list-end';
+import NavItem from 'components/section-nav/item';
```
Comment (rodrigoi) [inline client/devdocs/design/playground.jsx]: I've noticed that these changes are duplicated across many PRs (#24231, #24229, #24225 and #24224), maybe it's a good idea to extract those into it's own PR 😉

Comment (rodrigoi) [reply client/devdocs/design/playground.jsx]: Just found your response on #24223 (https://github.com/Automattic/wp-calypso/pull/24223#discussion_r182001584). Sorry for the duplication 🙂

File: client/components/user/docs/example.jsx (hunk 1 above)
Comment (andfinally) [reply client/components/user/docs/example.jsx]: Thanks @rodrigoi, I'll take a look!

File: client/devdocs/design/playground.jsx (hunk 2 above)
Comment (andfinally) [reply client/devdocs/design/playground.jsx]: Cool, I'll look at splitting those out.

File: client/components/user/docs/example.jsx (hunk 1 above)
Comment (scruffian) [reply client/components/user/docs/example.jsx]: Do you think it makes more sense to move this selector to the component itself?

Comment (andfinally) [reply client/components/user/docs/example.jsx]: Looking at other examples, the selector is typically used in a `connect` function, and it does seem to make sense to keep the data and the visual elements apart here, even though the visual element is just the `UserItem` component.

Comment (rodrigoi) [reply client/components/user/docs/example.jsx]: I think it's ok to `connect()( UserItemExample )`. I don't think that `UserItem` should be a connected component.

**Comment by alisterscott** [comment]
Are we still planning to merge this one? Need some more reviews?

**Comment by andfinally** [comment]
Hi @alisterscott, thanks, I forgot about this one! We could still merge it, so yes, it does need a review. I'll put a call out.

**Comment by flootr** [comment]
@andfinally LGTM 👍 you might want to do a rebase before merging as this PR is `1738 commits behind master`.

**Comment by andfinally** [comment]
Thanks a lot Markus! I'll get this rebased and merged.

**Comment by andfinally** [comment]
Hi @flootr, I had to add the example to two lists which were created after this PR, `component-examples.js` and `playground-scope.js`. Any chance you could take another look for me? I've rebased and done a force push of this branch.

**Comment by flootr** [comment]
@andfinally thanks, works perfect :)

**Comment by andfinally** [comment]
Thank you!

File: client/devdocs/design/playground-scope.js [hunk 3]
```diff
@@ -112,6 +112,7 @@ export TimeSince from 'components/time-since';
 export Timezone from 'components/timezone';
 export TokenFields from 'components/token-field';
 export Tooltip from 'components/tooltip';
+export UserItem from 'components/user/docs/example';
```
Comment (scruffian) [inline client/devdocs/design/playground-scope.js]: I don't think this should be the example here - it should be the actual component.

Comment (andfinally) [reply client/devdocs/design/playground-scope.js]: Thanks @scruffian, I'll correct that.

Comment (andfinally) [reply client/devdocs/design/playground-scope.js]: Hi @scruffian, I've hopefully addressed this in https://github.com/Automattic/wp-calypso/pull/26406 - I found it problematic getting the component to work in the playground, so removed it from the scope for now.
